dll/win32/imm32/imm.c

```diff
@@ -3608,29 +3608,67 @@ BOOL WINAPI ImmRegisterClient(PVOID ptr, /* FIXME: should point to SHAREDINFO st
     return TRUE;
 }
 
+/***********************************************************************
+ *		CtfImmIsTextFrameServiceDisabled(IMM32.@)
+ */
+BOOL WINAPI CtfImmIsTextFrameServiceDisabled(VOID)
+{
+    PTEB pTeb = NtCurrentTeb();
+    if (pTeb->Win32ClientInfo[0] & 0x400)
```

Comment:
A flag name will certainly be needed there.

Reply:
Also please use something like:
`((PW32CLIENTINFO)NtCurrentTeb()->Win32ClientInfo)->WhateverFieldIsThere`
(you need to check which header(s) you need to include for getting `PW32CLIENTINFO`; I know gdi32 includes it)

And, just do: `return (((PW32CLIENTINFO)NtCurrentTeb()->Win32ClientInfo)->WhateverFieldIsThere & 0x400);`

Reply:
OK.